rmarkdown/Rdemo.Rmd

```diff
@@ -317,31 +317,38 @@ Here we're just linking to the released `.png` files in the repo (clicking the i
 We could have also released the underlying STP count data and developed an analysis script directly on this dataset, provided it is non-disclosive. 
 However, typically only the most high-level aggregated datasets are suitable for public release. 
 
+### A note on dummy data
+
 We could also create something more sophisticated like a map, and include full STP names instead of codes.
-This requires a shape file, and to run successfully locally would also need matching dummy data. 
-See the [`study_definition_1_stppop_map.py`](https://github.com/opensafely/os-demo-research/blob/master/analysis/study_definition_1_stppop_map.py) study definition and [`analysis/1-plot-stppop_map.R`](https://github.com/opensafely/os-demo-research/blob/master/analysis/1-plot-stppop_map.R) R script to see how this would work.
+The outputs on real data look like this:
+
+[![](../released-ouput/plots/plot_stppop_bar_names.png "registration count by STP"){width="70%"}](https://github.com/opensafely/os-demo-research/blob/master/released-ouput/plots/plot_stppop_bar_names_v2.png)
+
+[![](../released-ouput/plots/plot_stppop_map.png "registration count by STP"){width="50%"}](https://github.com/opensafely/os-demo-research/blob/master/released-ouput/plots/plot_stppop_map_v2.png)
+
+As the script used to create the map uses a [shape file](https://github.com/opensafely/os-demo-research/blob/master/lib/STPshapefile.json), the dataset needs to contain the real STP codes so that the shape file matches the data. Running the script on dummy data where the STP codes are "STP1" "STP2" and "STP3", as above, wouldn't work. So the dummy data needs to usee real STP codes. 
 
-The outputs look like this:
+One way to do this is outlined in an alternative study definition, [`study_definition_1_stppop_v2.py`](https://github.com/opensafely/os-demo-research/blob/master/analysis/study_definition_1_stppop_v2.py), where the _expectations_ for the `stp` variable is based on real STP codes imported from a CSV file. 
```

Comment:
I wonder if we can make the fact there are alternatives clearer with a bit of renaming? 

The  `v1` and `v2` and `v3` stuff is a little confusing; partly because you're having to do some somersaults to work around having alternative execution paths with identical outputs.

Might it be easier to explain by having two different `project.yamls` that are almost exactly the same?

Or a commented-out expectations-only version that (in the comments) can say "uncomment this and comment out that in order to try the expectations-only path"?

I'd do a better job of checking this myself if I had my normal computer for checkouts etc, so I'm being a bit vaguer than I'd like.

Reply:
Yes this is difficult. It might even be better to have a completely new example to showcase custom dummy data, so that there's no `v2` and `v3` nonsense. 

Initially `_v2` was `_map`. We were keen to include a map in the demo because it's simple and striking. But it requires real STP codes which is just a little bit too complicated to digest on a first pass. 

It it possible to have two `project.yaml`s in the same repo? how does the CLI / job server choose which to use?

Maybe just commenting in/out the `dummy_data_file` line the the extractor action is good enough.

Reply:
> It it possible to have two project.yamls in the same repo? how does the CLI / job server choose which to use?
Right now, you can't direct the framework to use a different one. You'd just have to say in the docs (README & Rmd commentary & comments?) that to use the alternative one you need to rename them...

Not sure of the best answer but I _think_ I prefer either the rename or the comment-out-in-yaml approaches to the current one...

Reply:
I've redone it so that there's no awkward repeating of actions or the need to comment anything out. Basically the maps example is back the way it was before (with files and scripts using the `_maps` suffix) and I've created a completely new example for explaining custom dummy data